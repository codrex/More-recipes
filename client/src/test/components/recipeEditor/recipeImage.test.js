/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import {
  PureRecipeImage
} from '../../../components/pages/RecipeEditor/RecipeImage';

global.cloudinary = { openUploadWidget: jest.fn() };
const props = {
  image: '',
  updateImage: jest.fn()
};

describe('Recipe image component ', () => {
  let component;
  beforeEach(() => {
    component = mount(<PureRecipeImage {...props} />);
  });
  test('expected to match snapshot',
    () => {
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

  test('expected to match snapshot when loadImage button is clicked',
    () => {
      component.find('span.btn.bg-secondary').simulate('click');
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

  test('expected to match snapshot when component receive image as props',
    () => {
      component.setProps({ image: '{}' });
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
});
