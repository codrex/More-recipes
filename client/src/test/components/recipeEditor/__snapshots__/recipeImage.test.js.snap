// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe image component  render as expected when component is mounted 1`] = `
<div
  className="image-container"
>
  <h4
    className="items-header-text"
  >
    Recipe image
  </h4>
  <div
    className="display-4"
  >
    <Image
      className="recipe-card-img"
      cloudName="resycom"
      crop="fit"
      height="500"
      publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
      width="400"
    />
    <span
      className="btn bg-secondary"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Upload image
    </span>
  </div>
</div>
`;

exports[`Recipe image component  render as expected when component receive image as props 1`] = `
<div
  className="image-container"
>
  <h4
    className="items-header-text"
  >
    Recipe image
  </h4>
  <div
    className="display-4"
  >
    <Image
      className="recipe-card-img"
      cloudName="resycom"
      crop="fit"
      height="500"
      publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
      width="400"
    />
    <span
      className="btn bg-secondary"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Upload image
    </span>
  </div>
</div>
`;

exports[`Recipe image component  render as expected when loadImage button is click 1`] = `
<div
  className="image-container"
>
  <h4
    className="items-header-text"
  >
    Recipe image
  </h4>
  <div
    className="display-4"
  >
    <Image
      className="recipe-card-img"
      cloudName="resycom"
      crop="fit"
      height="500"
      publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
      width="400"
    />
    <span
      className="btn bg-secondary"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Upload image
    </span>
  </div>
</div>
`;
